main/display/bresenham.c

```diff
@@ -142,6 +142,56 @@ void plotLine(display_t* disp, int x0, int y0, int x1, int y1, paletteColor_t co
     }
 }
 
+void plotLineTranslate(display_t* disp, int x0, int y0, int x1, int y1, paletteColor_t col, int dashWidth, translateFn_t xTr, translateFn_t yTr)
```

Comment:
This comment applies to all the changes in `bresenham.c`, I'm really not a fan of copy/paste. What is the motivation behind using a function pointer as an argument, rather than an integer representation of the translation (or better yet, just translating `x0`, `x1`, `y0`, and `y1` before calling?

Reply:
Ah, so I think I originally copy-pasted these because I thought I had to use a function to do the translation/scaling operations without completely changing all the plotting logic, and then I didn't want to slow down all the nicely optimized code in `bresenham.c` with lots of extra function calls... But yeah I totally could have just used more parameters from the beginning.

I don't think I can just translate beforehand though, because I want to be able to draw scaled shapes (or at least `1 / (scale**2)` pixels of a scaled shape and I can handle drawing it `scale**2` times with all the offsets in between), and the translation won't necessarily be a multiple of the scale.

I think adding `xOffset`, `yOffset`, `xScale`, and `yScale` to each drawing function, plus a non-translating wrapper that passes `0, 0, 1, 1` for backwards compatibility would be OK and not slow down everyone else who doesn't need to scale.

Reply:
This passes my sniff test. @cnlohr may want to take a peek at what you've done here, but it's good enough for `main`